code/src/OSFramework/Feature/IValidationMark.ts

```diff
@@ -5,7 +5,8 @@ namespace OSFramework.Feature {
         invalidRows: Array<any>;
         clear(): void;
         errorMessage(rowNumber: number, binding: string): string;
-        isInvalid(rowNumber: number, binding: string): boolean;
+        isInvalidByRow(row: any): boolean;
+        isInvalidByRowNumber(rowNumber: number, binding: string): boolean;
```

Comment:
I would suggest to keep it as it was. Add only the **isInvalidRow** method.

Reply:
Here, should be like this **isInvalid(rowNumber: number, binding: string): boolean** (original method) and add **isInvalidRow(row: any): boolean** (new need)?